File: 203-remove-linked-list-elements/203-remove-linked-list-elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        if(head == null)
            return null;
        List<Integer>li = new ArrayList<>();
        ListNode temp = head;
        while(temp!=null)
        {
            if(temp.val!=val)
                li.add(temp.val);
            temp = temp.next;
        }
        ListNode root=null;
        temp = null;
        for(int i=0;i<li.size();i++)
        {
            if(root == null)
                root = temp = new ListNode((int)li.get(i));
            else{
                temp.next = new ListNode((int)li.get(i));
                    temp = temp.next;
            }
        }
        return root;
    }
}

File: 203-remove-linked-list-elements/203-remove-linked-list-elements.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        if(head == NULL)
            return NULL;
        vector<int>v ;//= new ArrayList<>();
        ListNode* temp = head;
        while(temp!=NULL)
        {
            if(temp->val!=val)
                v.push_back(temp->val);
            temp = temp->next;
        }
        ListNode* root=NULL;
        temp = NULL;
        for(int i=0;i<v.size();i++)
        {
            if(root == NULL)
                root = temp = new ListNode(v[i]);
            else{
                temp->next = new ListNode(v[i]);
                    temp = temp->next;
            }
        }
        return root;
    }
};